funcx_endpoint/tests/smoke_tests/test_s3_indirect.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+
+from funcx_endpoint.executors.high_throughput.funcx_worker import MaxResultSizeExceeded
+
+
+def large_result_producer(size: int) -> str:
+    return bytearray(size)
+
+
+def large_arg_consumer(data: str) -> int:
+    return len(data)
+
+
+test_cases = [200, 2000, 20000, 200000]
+
+
+@pytest.mark.parametrize("size", test_cases)
+def test_allowed_result_sizes(fx, endpoint, size):
+    """funcX should allow all listed result sizes which are under 512KB limit"""
+
+    future = fx.submit(large_result_producer, size, endpoint_id=endpoint)
+    x = future.result(timeout=60)
+    assert len(x) == size, "Result size does not match excepted size"
```

Comment:
```suggestion
    assert len(future.result(timeout=60)) == size
```

Because `pytest` will rewrite the bytecode for assertions to do deep assertion inspection and printing when asserts fail, I would tend to inline like this and not bother with the assert message in a case like this.

It's totally optional, but we don't gain information by including a fixed string in the failing case. In fact, including the call we want in the `assert` line will lead to pytest printing reprs for objects seen in that line, if there is a failure. Inlining can therefore sometimes make the testoutput better.